docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,804 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using an `Auth0` OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+With Quarkus, you can easily configure OIDC providers such as link:https://www.keycloak.org/documentation[Keycloak], link:https://developer.okta.com/[Okta], link:https://auth0.com/docs/[Auth0], and other xref:security-openid-connect-providers.adoc[well-known social OIDC and OAuth2 providers].
+
+Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with the https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.
+
+== Prerequisites
+
+
+To make it easier for you to complete this tutorial, review the following documentation before you begin:
+
+* link:https://auth0.com/docs/[Auth0 docs site]
+* xref:security-oidc-code-flow-authentication.adoc[Quarkus OpenID Connect authorization code flow mechanism for protecting web applications]
+* xref:security-oidc-bearer-token-authentication.adoc[Quarkus OpenID Connect (OIDC) Bearer token authentication]
+
+:sectnums:
+:sectnumlevels: 3
+
+== Create an Auth0 application
+
+Go to the `Auth0` dashboard and create a regular web application. 
+For example, create an `Auth0` application called `QuarkusAuth0`.
+
+image::auth0-create-application.png[Create Auth0 application]
+
+.Result
+Your Auth0 application gets created with a client ID, secret, and HTTPS-based domain.
+Make a note of these properties because you will need them to complete the Quarkus configuration in the next step.
+
+image::auth0-created-application.png[Created Auth0 application]
+
+Next, while still in the Auth0 dashboard, add some users to your application.
+
+image::auth0-add-user.png[Add Auth0 application users]
+
+Now that you have successfully created and configured your `Auth0` application, you are ready to start creating and configuring a Quarkus endpoint. 
+In the steps that follow, you will continue to configure and update your `Auth0` application as needed.
+
+== Create a Quarkus application
+
+Use the following Maven command to create a Quarkus RESTEasy Reactive application that can be secured with the Quarkus OIDC extension.
+
+:create-app-artifact-id: quarkus-auth0
+:create-app-extensions: resteasy-reactive,oidc
+include::{includes}/devtools/create-app.adoc[]
```

Comment:
This sequence creates a project without the generated code, so somehow it needs to be tweaked to have the starter code generated